src/CWL/CWL.fs

```diff
@@ -0,0 +1,116 @@
+namespace ARCtrl.CWL
+
+open DynamicObj
+open CWLTypes
+open Requirements
+open Inputs
+open Outputs
+open WorkflowSteps
+
+module CWLProcessingUnits =
+
+    type CWLToolDescription (
+            cwlVersion: string,
+            cls: Class,
+            outputs: Output [],
+            ?baseCommand: string [],
+            ?requirements: Requirement [],
+            ?hints: Requirement [],
+            ?inputs: Input [],
+            ?metadata: DynamicObj
+        ) =
+        inherit DynamicObj ()
+
+        let mutable _cwlVersion: string = cwlVersion
+        let mutable _class: Class = cls
+        let mutable _outputs: Output [] = outputs
+        let mutable _baseCommand: string [] option = baseCommand
+        let mutable _requirements: Requirement [] option = requirements
+        let mutable _hints: Requirement [] option = hints
+        let mutable _inputs: Input [] option = inputs
+        let mutable _metadata: DynamicObj option = metadata
+
+        member this.CWLVersion
+            with get() = _cwlVersion
+            and set(version) = _cwlVersion <- version
+
+        member this.Class
+            with get() = _class
+            and set(cls) = _class <- cls
+
+        member this.Outputs
+            with get() = _outputs
+            and set(outputs) = _outputs <- outputs
+
+        member this.BaseCommand
+            with get() = _baseCommand
+            and set(baseCommand) = _baseCommand <- baseCommand
+
+        member this.Requirements
+            with get() = _requirements
+            and set(requirements) = _requirements <- requirements
+
+        member this.Hints
+            with get() = _hints
+            and set(hints) = _hints <- hints
+
+        member this.Inputs
+            with get() = _inputs
+            and set(inputs) = _inputs <- inputs
+
+        member this.Metadata
+            with get() = _metadata
+            and set(metadata) = _metadata <- metadata
+
+    type CWLWorkflowDescription(
+        cwlVersion: string,
+        cls: Class,
```

Comment:
yes

Reply:
Then this Class property becomes kind of useless and (same as cwlVersion) possibly annoying for usage.

Reply:
possible solution: let both of them inherit from a base class with those 2 constructors, then pass the static one (cls in this case if i understand correctly) to the constructor of the base class, e.g. :
```fsharp
type A(a,b) = ...
type C(a) = inherit A(a,"cls1")
type D(a) = inherit A(a,"cls2")
```

this has the bonus that one can create sequences containing both `C` and `D` as they can be cast to the base class `A`

Reply:
This would be a possibility as an additional improvement, but I think removing this parameter from the constructor here makes sense either way. Your object does not need to be a 1to1 Copy of your fileformat. The field is there in the file to give information to your parser. But in your datamodel, this information is stored in the identity of the class. So having this parameter is duplication of information.